src/components/application_manager/test/commands/mobile/get_system_capability_request_test.cc

```diff
@@ -0,0 +1,246 @@
+#include "application_manager/test/include/application_manager/mock_hmi_capabilities.h"
+#include "gtest/gtest.h"
+
+namespace test {
+namespace components {
+namespace commands_test {
+namespace mobile_commands_test {
+namespace get_vehicle_data_tequest {
+
+using ::testing::_;
+using ::testing::Return;
+using ::testing::ReturnRef;
+namespace am = ::application_manager;
+namespace amt = application_manager_test;
+using am::commands::GetSystemCapabilityRequest;
+using am::commands::MessageSharedPtr;
+using am::event_engine::Event;
+namespace mobile_result = mobile_apis::Result;
+
+namespace {
+const uint32_t kConnectionKey = 2u;
+}  // namespace
+
+class GetSystemCapabilityRequestTest
+    : public CommandRequestTest<CommandsTestMocks::kIsNice> {
+ public:
+  MessageSharedPtr CreateMessageParams() {
+    MessageSharedPtr msg = CreateMessage(smart_objects::SmartType_Map);
+    (*msg)[am::strings::params][am::strings::connection_key] = kConnectionKey;
+    return msg;
+  }
+};
+
+TEST_F(GetSystemCapabilityRequestTest, Run_NotExistApp_AppNotRegistered) {
+  MockAppPtr mock_app;
+
+  EXPECT_CALL(app_mngr_, application(kConnectionKey))
+      .WillOnce(Return(mock_app));
+  MessageSharedPtr msg = CreateMessageParams();
+
+  SharedPtr<GetSystemCapabilityRequest> command(
+      CreateCommand<GetSystemCapabilityRequest>(msg));
+
+  EXPECT_CALL(
+      app_mngr_,
+      ManageMobileCommand(
+          MobileResultCodeIs(mobile_apis::Result::APPLICATION_NOT_REGISTERED),
+          _)).WillOnce(Return(true));
+
+  ASSERT_TRUE(command->Init());
+  command->Run();
+}
+
+TEST_F(GetSystemCapabilityRequestTest, Run_EmptyMessage_ExpectInvalidData) {
+  MockAppPtr mock_app = CreateMockApp();
+  EXPECT_CALL(app_mngr_, application(kConnectionKey))
+      .WillOnce(Return(mock_app));
+
+  MessageSharedPtr msg = CreateMessageParams();
+  SharedPtr<GetSystemCapabilityRequest> command(
+      CreateCommand<GetSystemCapabilityRequest>(msg));
+
+  EXPECT_CALL(app_mngr_,
+              ManageMobileCommand(
+                  MobileResultCodeIs(mobile_apis::Result::INVALID_DATA), _))
+      .WillOnce(Return(true));
+
+  ASSERT_TRUE(command->Init());
+  command->Run();
+}
+
+TEST_F(GetSystemCapabilityRequestTest, Run_NAVIGATION_SUCCESS) {
+  MockAppPtr mock_app = CreateMockApp();
+  SmartObject sm(smart_objects::SmartType_Map);
+  EXPECT_CALL(app_mngr_, application(kConnectionKey))
+      .WillOnce(Return(mock_app));
+  MessageSharedPtr msg = CreateMessageParams();
+  amt::MockHMICapabilities mock_hmi_cap;
+  (*msg)[am::strings::msg_params][am::strings::system_capability_type] =
+      mobile_apis::SystemCapabilityType::NAVIGATION;
+  SharedPtr<GetSystemCapabilityRequest> command(
+      CreateCommand<GetSystemCapabilityRequest>(msg));
+  EXPECT_CALL(app_mngr_, hmi_capabilities()).WillOnce(ReturnRef(mock_hmi_cap));
+  EXPECT_CALL(mock_hmi_cap, navigation_capability())
+      .WillRepeatedly(Return(&sm));
```

Comment:
@ValeriiMalkov Consider the possibility of using and `.Times (n)` along with `.WillRepeatedly`.